eZ/Publish/Core/Repository/RoleService.php

```diff
@@ -121,8 +126,386 @@ public function __construct(
     }
 
     /**
+     * Creates a new RoleDraft.
+     *
+     * @since 6.0
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException if the authenticated user is not allowed to create a role
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException if the name of the role already exists or if limitation of the
+     *                                                                        same type is repeated in the policy create struct or if
+     *                                                                        limitation is not allowed on module/function
+     * @throws \eZ\Publish\API\Repository\Exceptions\LimitationValidationException if a policy limitation in the $roleCreateStruct is not valid
+     *
+     * @param \eZ\Publish\API\Repository\Values\User\RoleCreateStruct $roleCreateStruct
+     *
+     * @return \eZ\Publish\API\Repository\Values\User\RoleDraft
+     */
+    public function createRoleDraft(APIRoleCreateStruct $roleCreateStruct)
+    {
+        if (!is_string($roleCreateStruct->identifier) || empty($roleCreateStruct->identifier)) {
+            throw new InvalidArgumentValue('identifier', $roleCreateStruct->identifier, 'RoleCreateStruct');
+        }
+
+        if ($this->repository->hasAccess('role', 'create') !== true) {
+            throw new UnauthorizedException('role', 'create');
+        }
+
+        try {
+            $existingRoleDraft = $this->loadRoleDraftByIdentifier($roleCreateStruct->identifier);
```

Comment:
@andrerom Strictly speaking I should check both, but on creation there will never actually be any conflicts on non-draft roles. When creating new roles an md5-based internal identifier is used, and when creating a draft from an existing role the existing role identifier is known to be unique.

When updating the role however, I certainly need to check both, and react differently in each conflict case.

How would you change the SPI? How about just adding the non-draft check here in the RoleService?